web/src/app/admin/admin-message-logs/AdminDebugMessagesLayout.tsx

```diff
@@ -77,50 +62,84 @@ export default function AdminDebugMessagesLayout(): JSX.Element {
     variables: { first: MAX_QUERY_ITEMS_COUNT },
   })
 
-  const [params, setParams] = useURLParams({
+  const [params] = useURLParams({
     search: '',
     start: '',
     end: '',
   })
 
-  const results = useFuse<DebugMessage>({
-    data:
-      data?.debugMessages.map((d: DebugMessage) => ({
-        ...d,
-        additionalKeys: {
-          filteredDestination: d.destination.replace('-', ''),
-        },
-      })) || [],
-    keys: [
-      'destination',
-      'userName',
-      'serviceName',
-      'status',
-      'additionalKeys.filteredDestination',
-    ],
-    search: params.search,
-    options: {
-      shouldSort: false,
-      showResultsWhenNoSearchTerm: true,
-      ignoreLocation: true,
-      useExtendedSearch: true,
-    },
-  })
+  if (error) return <GenericError error={error.message} />
+  if (loading && !data) return <Spinner />
 
   const startDT = params.start ? DateTime.fromISO(params.start) : null
   const endDT = params.end ? DateTime.fromISO(params.end) : null
 
-  const filteredResults = results.filter((result) => {
-    const createdAtDT = DateTime.fromISO(result.item.createdAt)
-    if (startDT && startDT > createdAtDT) return false
-    if (endDT && endDT < createdAtDT) return false
-    return true
-  })
+  const filteredData: DebugMessage[] = data?.debugMessages
```

Comment:
Can we move the filtering and bucketing into a separate hook and maybe pass it through the web worker?